### DISCUSSION TITLE
For a new project how can I tell poetry to use a given virtual environment?

### DISCUSSION THREAD

**Discussion by halloleo** [python-poetry/poetry#4698] [Q&A]
For a new Python project I normally let Poetry Create a new virtual environment:

I just do `poetry install` and poetry creates a new virtual environment in `./.venv` (I have set `virtualenvs.in-project = True`). All packages from the TOML file get installed in there. So far so great! :-) 

However for _some_ projects I want Poetry to use an existing venv under `/path/to/pre-exesiting/env` and install the packages into this venv.

To achieve this I tried `poetry env use /path/to/pre-exesiting/env/bin/python`, but Poetry _still_ creates a new environment in `./.venv`. :-(

**Comment by clintonroy** [comment]
Activate the environment, then poetry will use it.

So activate your environment (. .bin/activate or so) then change into the poetry project directory and poetry install.

It is documented at the top of https://python-poetry.org/docs/managing-environments/

**Reply by halloleo** [reply]
Thanks @clintonroy for chiming in!

I have tested this and the external environment is not written into Poetry's store: When I later come back to the project Poetry has forgotten about the venv and a simple `poetry add package` will create a new environment.

Is there a way to keep am external environment for Poetry associated with that project?

The documentation page you point to mentions under [Switching between environments](https://python-poetry.org/docs/managing-environments/#switching-between-environments) that I can switch venvs via `poetry` env use ..` but this does not work for me. :-( Maybe I misunderstand things here...
